library/Businessprocess/IcingaDbBackend.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Icinga\Module\Businessprocess;
+
+use Icinga\Module\Businessprocess\Common\IcingadbDatabase;
+use Icinga\Module\Icingadb\Model\Host;
+use Icinga\Module\Icingadb\Model\Service;
+use ipl\Orm\Compat\FilterProcessor;
+use ipl\Orm\Query;
+
+class IcingaDbBackend
+{
+    use MonitoringRestrictions;
+
+    use IcingadbDatabase;
+
+    /** @var BpConfig */
+    protected $config;
+
+    protected $backend;
+    public function __construct()
+    {
+        $this->backend = $this->getDb();
+    }
+
+    public function fetchHosts()
+    {
+
+        $hosts = Host::on($this->getDb())
+            ->orderBy('host.name');
+
+        $this->applyMonitoringRestriction($hosts);
+
+        return $hosts;
+    }
+
+    public function fetchServices($host)
+    {
+        $query = Service::on($this->backend)
```

Comment:
Rename `$query` to `$services`. Don't call `fetchAll` and remove the `var_dump`. You also don't need the `assembleSelect()` call. I guess this was only for debugging. Just return `$services` then.